### PR TITLE
cheat: init at 2.1.26

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
@Mic92 judging from the description of this package it seems it is an application, and not a Python library. Or is it both? If it is used as an application only, it needs to be moved outside of `python-packages.nix`.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -2084,6 +2084,26 @@ in modules // {
     };
   };
 
+  cheat = buildPythonPackage rec {
+    version = "2.1.26";
+    name = "cheat-${version}";
+
+    propagatedBuildInputs = with self; [ docopt pygments ];
+
+    src = pkgs.fetchFromGitHub {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: the package seems to be available on PyPI. For Python packages PyPI is the preferred source.

**Comment by Mic92** [comment]
fixed all concerns!

**Comment by Mic92** [comment]
fixed all concerns

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -6281,6 +6281,8 @@ in
 
   cgdb = callPackage ../development/tools/misc/cgdb { };
 
+  cheat = callPackage ../data/documentation/cheat { };
```
Comment (rasendubi) [inline pkgs/top-level/all-packages.nix]: `data` category is for data (not applications). A better place for `cheat` is `applications/office` or `applications/misc` (IMHO)

File: pkgs/data/documentation/cheat/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{ python35Packages, fetchurl, lib }:
```
Comment (rasendubi) [inline pkgs/data/documentation/cheat/default.nix]: It's better to not fix to python-3.5. `python3Packages` would be a better choice.

File: pkgs/data/documentation/cheat/default.nix [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+  propagatedBuildInputs = with python35Packages; [ docopt pygments ];
+
+  src = fetchurl {
+    url = "mirror://pypi/c/cheat/${name}.tar.gz";
+    sha256 = "0yilm9ba6ll9wzh20gj3lg9mnc50q95m6sqmjp2vcghwgipdixpm";
+  };
+
+  meta = {
+    description = "cheat allows you to create and view interactive cheatsheets on the command-line";
```
Comment (rasendubi) [inline pkgs/data/documentation/cheat/default.nix]: Please, add project home page to meta

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: I moved it to `applications/misc` now, it does definitely not belong to `application/office`. Its core asset are actually these short text files containing documentation like `manpages` or `gnome-user-docs`.

Comment (rasendubi) [inline pkgs/top-level/all-packages.nix]: `application/misc` is fine for me. Though, I should note that `application/office` doesn't mean application should deal with large nicely formatted documents and this kind of things.

In example, we currently have [todo.txt-cli](http://todotxt.com/) and [tudu](https://code.meskio.net/tudu/) in `application/office`, which both are small apps for managing todo lists. There are also [keepnote](http://keepnote.org/) and [eventlist](https://store.kde.org/content/show.php/Eventlist?content=107779), which are GUI apps for organizing other things.

**Comment by rasendubi** [comment]
LGTM

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: cheat is not a todo/note manager.

Comment (rasendubi) [inline pkgs/top-level/all-packages.nix]: IMHO, it's basically a note management software.

**Comment by Mic92** [comment]
anything left to do?

**Pull request merged**